arcgis-ios-sdk-samples/Layers/Display KML/DisplayKMLViewController.swift

```diff
@@ -0,0 +1,180 @@
+                // Display the error if one occurred
+                let alertController = UIAlertController(title: error.localizedDescription, message: nil, preferredStyle: .alert)
+                alertController.addAction(UIAlertAction(title: "OK", style: .cancel))
+                self.present(alertController, animated: true, completion: nil)
+            }
+        }
+        
+    }
+    
+    private func changeSourceToURL() {
+        /// A URL of a remote KML file
+        let kmlDatasetURL = URL(string: "https://www.wpc.ncep.noaa.gov/kml/noaa_chart/WPC_Day1_SigWx.kml")!
+        let kmlDataset = AGSKMLDataset(url: kmlDatasetURL)
+        /// A KML layer created from a remote KML file
+        let kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+        display(kmlLayer: kmlLayer)
+    }
+    
+    private func changeSourceToLocalFile() {
+        /// A dataset created by referencing the name of a KML file in the app bundle
+        let kmlDataset = AGSKMLDataset(name: "US_State_Capitals")
+        /// A KML layer created from a local KML file
+        let kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+        display(kmlLayer: kmlLayer)
+    }
+    
+    private func changeSourceToPortalItem() {
+        let portal = AGSPortal.arcGISOnline(withLoginRequired: false)
+        /// A remote KML portal item
+        let portalItem = AGSPortalItem(portal: portal, itemID: "9fe0b1bfdcd64c83bd77ea0452c76253")
+        /// A KML layer created from an ArcGIS Online portal item
+        let kmlLayer = AGSKMLLayer(item: portalItem)
+        display(kmlLayer: kmlLayer)
+    }
+    
+}
+
+extension DisplayKMLViewController: UIPopoverPresentationControllerDelegate {
+    
+    func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
+        // prevent the popover from displaying fullscreen in narrow contexts
+        return .none
+    }
+}
+
+class DisplayKMLDataSourcesTableViewController: UITableViewController {
+    
+    fileprivate var initialSelectedIndex = 0
+    fileprivate var selectionHandler: ((KMLDataSource)->Void)?
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
+        if indexPath.row == initialSelectedIndex{
+            cell.accessoryType = .checkmark
+            tableView.selectRow(at: indexPath, animated: false, scrollPosition: .none)
+        }
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        // get the newly selected cell
+        let selectedCell = tableView.cellForRow(at: indexPath)!
+        
+        // add a checkmark to the selected cell
+        selectedCell.accessoryType = .checkmark
+        
+        // change the displayed KML source based on the selected cell
+        let displayedSource = KMLDataSource(rawValue: indexPath.row)!
```

Comment:
I try to avoid implicitly unwrapping optionals. I know this will *never* fail but I try to avoid `!` whenever possible. Can you `guard let` this instead? This won't depend on my PR approval, just specifying my preference.